src/applications/travel-pay/components/complex-claims/pages/ChooseExpenseType.jsx

```diff
@@ -1,33 +1,67 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { useNavigate, useParams } from 'react-router-dom-v5-compat';
+import { useSelector, useDispatch } from 'react-redux';
 import {
   VaRadio,
   VaButtonPair,
 } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
-import { EXPENSE_TYPES } from '../../../constants';
+import { EXPENSE_TYPES, EXPENSE_TYPE_KEYS } from '../../../constants';
+import { getClaimDetails } from '../../../redux/actions';
 
 const ChooseExpenseType = () => {
   const navigate = useNavigate();
+  const dispatch = useDispatch();
   const { apptId, claimId } = useParams();
   const [selectedExpenseType, setSelectedExpenseType] = useState('');
   const [showError, setShowError] = useState(false);
+  const [mileageError, setMileageError] = useState(false);
+
+  // Get claim data from Redux
+  const { data, isLoading } = useSelector(
+    state => state.travelPay.claimDetails,
+  );
+  const claim = data[claimId];
+
+  // Fetch claim details
+  useEffect(
+    () => {
+      if (claimId && !claim && !isLoading) {
+        dispatch(getClaimDetails(claimId));
+      }
+    },
+    [claimId, claim, isLoading, dispatch],
+  );
```

Comment:
I believe we shouldn't need this here since it's already happening in the `ComplexClaimSubmitFlowWrapper` component. You may want the `selectComplexClaim` selector for the version of claim details we're getting for complex claims:
```
const { data: claimDetails = {} } = useSelector(selectComplexClaim);
```